statistics/handle/handle.go

```diff
@@ -221,6 +221,118 @@ func (h *Handle) Update(is infoschema.InfoSchema) error {
 	return nil
 }
 
+// GlobalStats is used to store the statistics contained in the global-level stats
+// which is generated by the merge of partition-level stats.
+// It will both store the column stats and index stats.
+// In the column statistics, the variable `num` is equal to the number of columns in the partition table.
+// In the index statistics, the variable `num` is always equal to one.
+type GlobalStats struct {
+	Num   int
+	Count int64
+	Hg    []*statistics.Histogram
+	Cms   []*statistics.CMSketch
+	TopN  []*statistics.TopN
+}
+
+// MergePartitionStats2GlobalStats merge the partition-level stats to global-level stats based on the tableID.
+func (h *Handle) MergePartitionStats2GlobalStats(is infoschema.InfoSchema, physicalID int64, isIndex int, idxID int64) (globalStats *GlobalStats, succ bool) {
```

Comment:
It's better to add an `error` to its returned values and return all errors in this function.